packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -1514,6 +1516,63 @@ Update them to point to this script instead?`,
 			});
 		});
 
+		it("should error on routes with paths if experimental assets are present", async () => {
+			writeWranglerToml({
+				routes: [
+					"simple.co.uk/path",
+					"simple.co.uk/path/*",
+					"simple.co.uk/",
+					"simple.co.uk/*",
+					"simple.co.uk",
+					{ pattern: "route.co.uk/path", zone_id: "asdfadsf" },
+					{ pattern: "route.co.uk/path/*", zone_id: "asdfadsf" },
+					{ pattern: "route.co.uk/*", zone_id: "asdfadsf" },
+					{ pattern: "route.co.uk/", zone_id: "asdfadsf" },
+					{ pattern: "route.co.uk", zone_id: "asdfadsf" },
+					{ pattern: "custom.co.uk/path", custom_domain: true },
+					{ pattern: "custom.co.uk/*", custom_domain: true },
+					{ pattern: "custom.co.uk", custom_domain: true },
+				],
+			});
+			writeWorkerSource();
+			writeAssets([{ filePath: "asset.txt", content: "Content of file-1" }]);
+
+			await expect(runWrangler(`deploy --experimental-assets="assets"`)).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Invalid Routes:
+				simple.co.uk/path:
+				Workers which have static assets cannot be routed on a URL which has a path component. Update the route to replace /path with /*
+
+				simple.co.uk/path/*:
+				Workers which have static assets cannot be routed on a URL which has a path component. Update the route to replace /path/* with /*
+
+				simple.co.uk/:
+				Update the route to end with /*
```

Comment:
I'm slightly worried about this error message. In non-assets Workers this is a valid route, right?
So the message should be similar to those above, no?
Currently I would be a bit confused to see this message.